frontend/BoardGUI.py

```diff
@@ -37,29 +41,101 @@ def build_board(self):
         self.draw_board()
 
     def draw_piece(self, turn, position):
-        if turn == Turn.BLACK:
+        if turn == Color.BLACK:
             self.game.screen.blit(self.black_piece_image, position)
         else:
             self.game.screen.blit(self.white_piece_image, position)
 
+    def remove_piece(self, row, col):
+        # Remove the piece from the internal structure
+        if (row, col) in self.pieces_on_board:
+            del self.pieces_on_board[(row, col)]
+
+    
+    def handle_mill(self,row, col):
+        if (row, col) in self.game_manager.removable_pieces:
+            # Valid piece selected to remove
+            print(f"Removing opponent's piece at ({row}, {col})")
+            self.remove_piece(row, col)  # Remove the piece from the board
+            self.draw_board()  # Redraw the board after removal
```

Comment:
Can we just remove the piece at the position instead of redrawing the whole board?